src/coreComponents/virtualElement/VirtualElementBase.hpp

```diff
@@ -37,21 +40,71 @@ class VirtualElementBase
   virtual ~VirtualElementBase() = default;
 
-  /// @brief Compute VEM projectors on the geometry.
-  virtual void computeProjectors( MeshLevel const &, localIndex const &,
-                                  localIndex const &, localIndex const & ) = 0;
+  /**
+   * @brief Get the shape function projected derivatives at a given quadrature point.
+   * @details The output contains the integral mean of derivatives.
+   * @param q The quadrature point index.
+   * @param gradN Return array of the shape function projected derivatives. Size will be @ref
+   * getNumSupportPoints() x 3.
+   */
+  template< typename LEAF >
+  // GEOSX_HOST_DEVICE
+  static real64 getGradN( localIndex const q,
+                          real64 ( & gradN )[LEAF::maxSupportPoints][3] )
+  {
+    return LEAF::calcGradN( q, gradN );
+  }
 
   /**
-   * @brief Virtual getter for the number of quadrature points per element.
-   * @return The number of quadrature points per element.
+   * @brief Get a value of the stabilization matrix.
+   * @param iBasisFunction The row index.
+   * @param jBasisFunction The column index.
+   * @return The requested value.
    */
-  virtual localIndex getNumQuadraturePoints() const = 0;
+  template< typename LEAF >
+  // GEOSX_HOST_DEVICE
+  static real64 getStabilizationValue( localIndex const iBasisFunction,
+                                       localIndex const jBasisFunction
+                                       )
+  { return LEAF::calcStabilizationValues( iBasisFunction, jBasisFunction ); }
 
   /**
-   * @brief Virtual getter for the number of support points per element.
-   * @return The number of support points per element.
+   * @brief Get the shape function projections at a given quadrature point.
+   * @details The output contains the integral mean of functions
+   * @param q The quadrature point index.
+   * @param gradN Return array of the shape function projections. Size will be @ref getNumSupportPoints().
    */
-  virtual localIndex getNumSupportPoints() const = 0;
+  template< typename LEAF >
+  // GEOSX_HOST_DEVICE
+  void getN( localIndex const q,
+             real64 ( & N )[LEAF::maxSupportPoints] )
+  {
+    return LEAF::calcN( q, N );
+  }
+
+  /**
+   * @brief Get the integration weight for a quadrature point.
+   * @param q Index of the quadrature point.
+   * @return The weight.
+   */
+  template< typename LEAF >
+  GEOSX_HOST_DEVICE
+  real64 getTransformedQuadratureWeight( localIndex const q ) const
+  {
+    return LEAF::getTransformedQuadratureWeight( q );
+  }
+
+  // /**
+  //  * @brief Virtual getter for the number of quadrature points per element.
+  //  * @return The number of quadrature points per element.
+  //  */
+  // virtual localIndex getNumQuadraturePoints() const = 0;
+
+  // /**
+  //  * @brief Virtual getter for the number of support points per element.
+  //  * @return The number of support points per element.
+  //  */
+  // virtual localIndex getNumSupportPoints() const = 0;
```

Comment:
Maybe remove if you don't plan to use these getters.

Reply:
I added an implementation of the methods, templatized linke to the other ones in the class.